### ISSUE TITLE
Inconsistent return values

### ISSUE THREAD

**Issue by leofeyer** [contao/core#8401]
Look at the following methods:

* https://github.com/contao/core/blob/master/system/modules/faq/models/FaqModel.php#L131-L147

```php
/**
 * Find a published FAQ from one or more categories by its ID or alias
 *
 * …
 *
 * @return \FaqModel|null The model or null if there is no FAQ
 */
public static function findPublishedByParentAndIdOrAlias($varId, $arrPids, array $arrOptions=array())
{
	// …

	return static::findOneBy($arrColumns, array((is_numeric($varId) ? $varId : 0), $varId), $arrOptions);
}
```

* https://github.com/contao/core/blob/master/system/modules/calendar/models/CalendarEventsModel.php#L199-L215

```php
/**
 * Find a published event from one or more calendars by its ID or alias
 *
 * …
 *
 * @return \CalendarEventsModel|null The model or null if there is no event
 */
public static function findPublishedByParentAndIdOrAlias($varId, $arrPids, array $arrOptions=array())
{
	// …

	return static::findOneBy($arrColumns, array((is_numeric($varId) ? $varId : 0), $varId), $arrOptions);
}
```

* https://github.com/contao/core/blob/master/system/modules/news/models/NewsModel.php#L167-L193

```php
/**
 * Find published news items by their parent ID and ID or alias
 *
 * …
 *
 * @return \NewsModel|null The model or null if there are no news
 */
public static function findPublishedByParentAndIdOrAlias($varId, $arrPids, array $arrOptions=array())
{
	// …

	return static::findBy($arrColumns, array((is_numeric($varId) ? $varId : 0), $varId), $arrOptions);
}
```

In the news module (third example), a collection is returned (`static::findBy()`), whereas in the other methods, a model is returned (`static::findOneBy()`). This causes e.g. problems in the [changelanguage](https://github.com/terminal42/contao-changelanguage/blob/develop/library/Terminal42/ChangeLanguage/EventListener/Navigation/AbstractNavigationListener.php#L33) bundle, if `findCurrent()` returns a collection instead of a model.

@contao/developers What to do about this? The phpDoc comment says "items" but the return tag says "model".

**Comment by aschempp** [comment]
Well there can always only be one result, right? I would consider this a bug and change the call to `findOneBy`. It should not make a difference for retrieving properties.

**Comment by Toflar** [comment]
Yes, it's a bug, I agree.

**Comment by leofeyer** [comment]
Fixed in 114cf47f4bf6c2176a9d4beff0d270d5a25d0706 then.
